kratos/includes/data_communicator.h

```diff
@@ -0,0 +1,446 @@
+//    |  /           |
```

Comment:
no communication for serialized objects? (that would be a communication for a std::string)

Reply:
I am replicating the existing capabilities of `MPICommunicator` and the `mpi` Python interface for this PR. We can add more as needed, but I thought the branch was getting big enough as is. I'm adding a note in the MPICore project to remind myself to add string communication.